Source/ARTRealtime.m

```diff
@@ -567,12 +613,43 @@ - (void)transportReconnectWithHost:(NSString *)host {
     [self.transport connect];
 }
 
-- (void)transportReconnectWithRenewedToken {
-    _renewingToken = true;
+- (void)transportReconnectWithTokenDetails:(ARTTokenDetails *)tokenDetails error:(NSError *)error {
+    if (error && (self.options.authUrl || self.options.authCallback)) {
+        [self.connection setErrorReason:[ARTErrorInfo createWithCode:ARTCodeErrorAuthConfiguredProviderFailure message:error.localizedDescription]];
+        return;
+    }
     [_transport close];
     _transport = [[_transportClass alloc] initWithRest:self.rest options:self.options resumeKey:_transport.resumeKey connectionSerial:_transport.connectionSerial];
     _transport.delegate = self;
-    [_transport connectForcingNewToken:true];
+    [_transport connectWithToken:tokenDetails.token error:error];
+}
+
+- (void)transportReconnectWithRenewedToken {
```

Comment:
Unless I'm missing something, all this logic will only apply when `transportReconnectWithRenewedToken` is called, ie. when an AUTH or ERROR is received, but not when e. g. first connection or manually calling `authorize`. The spec says "If an attempt by the realtime client library to authenticate is made using...", so _every_ auth attempt should follow this logic. I'd say this logic should be inside `Auth.authorize` itself, don't you think?